Intermittent test failure in ProxyPublishConsumeTest

```
socketTest(com.yahoo.pulsar.websocket.proxy.ProxyPublishConsumeTest)  Time elapsed: 3.029 sec  <<< FAILURE!
java.lang.AssertionError: null: lists don't have the same size expected [0] but found [10]
	at com.yahoo.pulsar.websocket.proxy.ProxyPublishConsumeTest.socketTest(ProxyPublishConsumeTest.java:100)

```

--- a/pulsar-broker/src/test/java/com/yahoo/pulsar/websocket/proxy/SimpleConsumerSocket.java
+++ b/pulsar-broker/src/test/java/com/yahoo/pulsar/websocket/proxy/SimpleConsumerSocket.java
@@ -39,7 +39,7 @@ public class SimpleConsumerSocket {
     private static final String X_PULSAR_MESSAGE_ID = "messageId";
     private final CountDownLatch closeLatch;
     private Session session;
-    private ArrayList<String> consumerBuffer = new ArrayList<String>();
+    private final ArrayList<String> consumerBuffer;
 
     public SimpleConsumerSocket() {
         this.closeLatch = new CountDownLatch(1);
@@ -65,7 +65,7 @@ public class SimpleConsumerSocket {
     }
 
     @OnWebSocketMessage
-    public void onMessage(String msg) throws JsonParseException, IOException {
+    public synchronized void onMessage(String msg) throws JsonParseException, IOException {
         JsonObject message = new Gson().fromJson(msg, JsonObject.class);
         JsonObject ack = new JsonObject();
         String messageId = message.get(X_PULSAR_MESSAGE_ID).getAsString();
@@ -83,7 +83,7 @@ public class SimpleConsumerSocket {
         return this.session;
     }
 
-    public ArrayList<String> getBuffer() {
+    public synchronized ArrayList<String> getBuffer() {
         return consumerBuffer;
     }
 

--- a/pulsar-broker/src/test/java/com/yahoo/pulsar/websocket/proxy/SimpleProducerSocket.java
+++ b/pulsar-broker/src/test/java/com/yahoo/pulsar/websocket/proxy/SimpleProducerSocket.java
@@ -42,7 +42,7 @@ public class SimpleProducerSocket {
 
     private final CountDownLatch closeLatch;
     private Session session;
-    private ArrayList<String> producerBuffer;
+    private final ArrayList<String> producerBuffer;
 
     public SimpleProducerSocket() {
         this.closeLatch = new CountDownLatch(1);
@@ -78,7 +78,7 @@ public class SimpleProducerSocket {
     }
 
     @OnWebSocketMessage
-    public void onMessage(String msg) throws JsonParseException {
+    public synchronized void onMessage(String msg) throws JsonParseException {
         JsonObject ack = new Gson().fromJson(msg, JsonObject.class);
         producerBuffer.add(ack.get("messageId").getAsString());
     }
@@ -91,7 +91,7 @@ public class SimpleProducerSocket {
         return this.session;
     }
 
-    public ArrayList<String> getBuffer() {
+    public synchronized ArrayList<String> getBuffer() {
         return producerBuffer;
     }
 

